README.md

```diff
@@ -1909,6 +1909,36 @@ f()for i in "$@"; do echo "$i"; done
 }
 ```
 
+Important note: using the `&&` syntax in a function can lead to
+unexpected results.  For example:
+
+```shell
+func_example1() {
+    if [ $1 -gt 11 ]; then
+        return
+    fi
+}
+
+func_example2() {
+    [ $1 -gt 11 ] && return
```

Comment:
In this use case one should explicitly supply exit code to `return`, so that it doesn't use last command's (`test`) exit status by default. And this is what the difference between the two constructions is. So that's kind of NOT unexpected result. See `help return`. I think this should be mentioned in this snippet to clarify what the "unexpected result" means, alongside the proper usage.